src/git/tag.go

```diff
@@ -42,8 +43,9 @@ var MatchAllTags = regexp.MustCompile("")
 
 func NewRepoTagsGetter(workDir string, opts ...TagOptionFunc) (*RepoTagsGetter, error) {
 	s := &RepoTagsGetter{
-		workDir: workDir,
-		match:   MatchAllTags,
+		workDir:       workDir,
+		match:         MatchAllTags,
+		commitsGetter: NewRepoCommitsGetter(workDir),
 	}
```

Comment:
Since You are composing the RepTagsGetter with a commitsGetter interface, then I would inject the NewRepoCommitsGetter from the call and change the signature to b eNewRepoTagsGetter(workDir string, commitsGetter CommitsGetter, opts ...TagOptionFunc).

Reply:
Makes sense!